neo3/wallet/wallet.py

```diff
@@ -0,0 +1,202 @@
+                "key": {"type": "string"},
+                "contract": {"type": ""},
+                "extra": {"type": ["object", "null"],
+                          "properties": {},
+                          "additionalProperties": True}
+            },
+            "required": ["address", "label", "is_default", "lock", "key", "contract", "extra"]
+
+        },
+        "scrypt_parameters": {
+            "type": "object",
+            "properties": {
+                "n": {"type": "integer"},
+                "r": {"type": "integer"},
+                "p": {"type": "integer"},
+                "length": {"type": "integer"}
+            },
+            "required": ["n", "r", "p"]
+        }
+    }
+}
+
+
+class Wallet(IJson):
+
+    _wallet_version = '3.0'
+
+    def __init__(self,
+                 path: str,
+                 version: str,
+                 scrypt: ScryptParameters,
+                 name: Optional[str] = None,
+                 accounts: List[Account] = None,
+                 extra: Optional[Dict[Any, Any]] = None):
+        """
+        Args:
+            path: the JSON's path
+            name: a label that the user has given to the wallet
+            version: the wallet's version, must be equal or greater then 3.0
+            scrypt: a ScryptParameters object which describes the parameters of the SCrypt algorithm used for encrypting
+                    and decrypting the private keys in the wallet.
+            accounts: an array of Account objects which describe the details of each account in the wallet.
+            extra: an object that is defined by the implementor of the client for storing extra data. This field can be
+                   None.
+        """
+
+        self.path = path
+        self.name = name
+        self.version = version
+        self.scrypt = scrypt
+        self.accounts = accounts if accounts is not None else []
+        self.extra = extra
+
+    @classmethod
+    def default(cls, path: str, name: Optional[str] = None) -> Wallet:
+        """
+        Create a new Wallet with the default settings.
+
+        Args:
+            path: the JSON's path.
+            name: the Wallet name.
+        """
+        return cls(path=path,
+                   name=name,
+                   version=cls._wallet_version,
+                   scrypt=ScryptParameters.default(),
+                   accounts=[],
+                   extra=None)
+
+    @classmethod
+    def new_wallet(cls, location: str):
+        filepath, extension = os.path.splitext(location)
+        if len(extension) == 0:
+            location += '.json'
+        elif extension != '.json':
+            # won't create if the file is not a json file
+            return None
+
+        try:
+            # creates the wallet file. If it already exists, returns None instead
+            file = open(location, 'x')
+            file.close()
+        except FileExistsError:
+            return None
```

Comment:
I'm not entirely sure if this will not leak on exception. I see multiple solutions
1. move `file.close()` to a `finally` block
2. open with a context manager `with open(location,'x'): pass`
3. use `Path(location).is_file()` and `touch()` from the `pathlib` module